src/averageAges.js

```diff
@@ -37,7 +38,12 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const arrayOfWoman = people.filter(person => withChildren
+    ? person.sex === 'f' && people.some(({ mother }) => mother === person.name)
+    : person.sex === 'f');
```

Comment:
a little hint here, move common statement out of condition
```suggestion
  const arrayOfWoman = people.filter(({ sex, name })=> 
     sex === 'f' && (withChildren ? people.some(({ mother }) => mother === name) : true)
  );
```
smarter way ⬇️
```suggestion
  const arrayOfWoman = people.filter(person => 
     person.sex === 'f' && (!withChildren || people.some(({ mother }) => mother === person.name) )
  );
```